### ISSUE TITLE
Difficulties on Remote Connection

### ISSUE THREAD

**Issue by alexpvieira** [cobisja/tad-php#33]
Hello. I'm trying to access a remote device but can't get the connection to work. It was working fine when the device and the PHP server where sharing the same network.

I've tried to open the ports 80 and 4370 on the server but got the following error:

`'TADPHP\Exceptions\ConnectionError' with message 'Imposible iniciar conexión con dispositivo obraargentina.ddns.net' in /opt/bitnami/apps/sistema-tecnofort/htdocs/tad/lib/TAD.php:408 Stack trace: #0 /opt/bitnami/apps/sistema-tecnofort/htdocs/tad/lib/TAD.php(271): TADPHP\TAD->check_for_connection() #1 /opt/bitnami/apps/sistema-tecnofort/htdocs/tad.php(29): TADPHP\TAD->__call('get_date', Array) #2 /opt/bitnami/apps/sistema-tecnofort/htdocs/tad.php(29): TADPHP\TAD->get_date() #3 {main}`

Did you get this working with a remote device like that?

**Comment by eizix** [comment]
make sure you use the correct ip address for the device network .. you may use the website: whatismyip.com
and add that ip in the setting for tad

**Comment by alexpvieira** [comment]
Yes, I'm using the correct IP. Actualy I'm using a DDNS but I've tried with the internet IP as well.

**Comment by alexpvieira** [comment]
@cobisja any input on this? Thank you.

**Comment by cobisja** [comment]
Hi,

Telling the truth, I did not test the library with a device connected to an
outside network, but I think that should not be a problem.

Are you sure your device is reachable from Internet?

Try to make a simple:

*ping obraargentina.ddns.net <http://obraargentina.ddns.net>*

To see if you can get an answer. If not, try to check your network
accessibility (firewalls, web port closed, etc).

See you,

2017-08-28 9:41 GMT-04:00 alexpvieira <notifications@github.com>:
